apps/api/src/modules/notifications/dtos/providers/awsSes-data.dto.ts

```diff
@@ -0,0 +1,41 @@
+import { Type } from 'class-transformer';
+import { IsNotEmpty, IsOptional, IsString, ValidateIf, ValidateNested } from 'class-validator';
+import {
+  AttachmentValidation,
+  CreateNotificationAttachmentDto,
+} from '../create-notification-attachment.dto';
+
+export class AwsSesDataDto {
+  @IsNotEmpty()
+  from: string | string[];
+
+  @IsNotEmpty()
+  to: string | string[];
+
+  @IsOptional()
+  cc?: string | string[];
+
+  @IsOptional()
+  bcc?: string | string[];
+
+  @IsNotEmpty()
+  @IsString()
+  subject: string;
+
+  @IsNotEmpty({ message: 'HTML or text must be provided' })
+  @ValidateIf((obj) => !obj.html, { message: 'HTML or text must be provided' })
+  text: string;
+
+  @IsNotEmpty({ message: 'HTML or text must be provided' })
+  @ValidateIf((obj) => !obj.text, { message: 'HTML or text must be provided' })
+  html: string;
+
+  @IsOptional()
+  replyTo?: string | string[];
+
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => CreateNotificationAttachmentDto)
+  @AttachmentValidation()
+  attachments: CreateNotificationAttachmentDto[];
```

Comment:
Check if  not adding + @IsString() still gives bad request?
 from: string | string[]; => Add non string value and checkif request is immediately dewnied like with @IsString()

Reply:
Added @IsString

Reply:
@kshitij-k-osmosys  Did you check if it takes both array of strings and single string?

Reply:
WIth the @IsString check it only accepts single string

Reply:
@kshitij-k-osmosys 
Try adding a custom validator in order to reuse(we could rely on ValidateIf too but since we want to reuse better to have reusable validator).

Generated decorator with ChatGPT.

```ts
import { registerDecorator, ValidationOptions, ValidationArguments } from 'class-validator';

export function IsStringOrStringArray(validationOptions?: ValidationOptions) {
  return function (object: Object, propertyName: string) {
    registerDecorator({
      name: 'isStringOrStringArray',
      target: object.constructor,
      propertyName: propertyName,
      options: validationOptions,
      validator: {
        validate(value: any, args: ValidationArguments) {
          if (typeof value === 'string') {
            return true; // Valid if it's a string
          }
          if (Array.isArray(value) && value.every(item => typeof item === 'string')) {
            return true; // Valid if it's an array of strings
          }
          return false; // Invalid otherwise
        },
        defaultMessage(args: ValidationArguments) {
          return `${args.property} must be either a string or an array of strings`;
        }
      }
    });
  };
}

```

Reply:
```ts
export class YourDto {

  @IsStringOrStringArray({ message: 'The "to" field must be either a string or an array of strings' })
  to: string | string[];

  @IsStringOrStringArray({ message: 'The "cc" field must be either a string or an array of strings' })
  cc: string | string[];

  @IsStringOrStringArray({ message: 'The "bcc" field must be either a string or an array of strings' })
  bcc: string | string[];

  // You can now use this for all similar fields
}

```

Reply:
Added custom validator IsStringOrStringArray